models/au_geoscape_gnaf/code/upload.py

```
@@ -0,0 +1,248 @@
+
+def _bq_client():
+    return bigquery.Client(project=BILLING_PROJECT)
+
+
+def ensure_staging_dataset():
+    """Create the staging dataset (location US) if it does not exist."""
+    bq = _bq_client()
+    ds_id = f"{BILLING_PROJECT}.{STAGING_DATASET}"
+    try:
+        bq.get_dataset(ds_id)
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)